src/app/loadout-drawer/loadout-drawer-reducer.ts

```diff
@@ -681,3 +662,166 @@ export function removeArtifactUnlock(mod: number): LoadoutUpdateFunction {
     }
   });
 }
+
+/** Randomize the subclass and subclass configuration */
+export function randomizeLoadoutSubclass(
+  defs: D1ManifestDefinitions | D2ManifestDefinitions,
+  store: DimStore
+): LoadoutUpdateFunction {
+  return (loadout) => {
+    const newSubclass = _.sample(
+      store.items.filter(
+        (item) => item.bucket.hash === BucketHashes.Subclass && itemCanBeInLoadout(item)
+      )
+    );
+
+    if (!newSubclass || !defs.isDestiny2()) {
+      return loadout;
+    }
```

Comment:
This prevents randomize-subclass from working on D1. Probably you only want to skip the socketOverrides.